### PR TITLE
[Python] Make Logging Filtering global

### PR DESCRIPTION
**Pull request by agners**
Log filtering is a global affair, so move the filtering functions to a global context too. This allows to setup logging before creating any device controller. It aligns with how the SDK treats logging.

Also enable log filtering for the Linux platform to make the functionality useful on Linux.

### CODE REVIEW

File: src/platform/Linux/CHIPPlatformConfig.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ using CHIP_CONFIG_PERSISTED_STORAGE_KEY_TYPE = const char *;
 #endif // CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS
 
 #ifndef CHIP_LOG_FILTERING
-#define CHIP_LOG_FILTERING 0
+#define CHIP_LOG_FILTERING 1
```
Comment (yunhanw-google) [inline src/platform/Linux/CHIPPlatformConfig.h:60]: this would enable log filtering for all linux example/app, can we just enable it for python only?

**Review by yunhanw-google — CHANGES_REQUESTED** [review]

File: src/platform/Linux/CHIPPlatformConfig.h (hunk 1 above)
Comment (yunhanw-google) [reply src/platform/Linux/CHIPPlatformConfig.h:60]: maybe defined from controller/python/BUILD.gn?

Comment (agners) [reply src/platform/Linux/CHIPPlatformConfig.h:60]: I was thinking the same at first, I tried by adding a define in `src/controller/python/BUILD.gn`:

```gn
  defines = [ "CHIP_LOG_FILTERING=1" ]
```

However, that failed at compile time:

```
FAILED: obj/src/controller/python/chip/_ChipDeviceCtrl.so obj/src/controller/python/chip/_ChipDeviceCtrl.map                                                                                                                                                                                                                  
g++ -O0 -fPIC -Werror -Wl,--fatal-warnings -Wl,-z,defs -fdiagnostics-color -Wl,--gc-sections -Wl,-Map,obj/src/controller/python/chip/_ChipDeviceCtrl.map -Wl,--cref @obj/src/controller/python/chip/_ChipDeviceCtrl.so.rsp -o obj/src/controller/python/chip/_ChipDeviceCtrl.so -shared                                       
/usr/bin/ld: obj/src/controller/python/_ChipDeviceCtrl.ChipDeviceController-ScriptBinding.cpp.o: in function `pychip_DeviceController_GetLogFilter':                                                                                                                                                                          
/home/sag/projects/project-chip/connectedhomeip-v1.2/out/python_lib/../../src/controller/python/ChipDeviceController-ScriptBinding.cpp:357:(.text.pychip_DeviceController_GetLogFilter+0x5): undefined reference to `chip::Logging::GetLogFilter()'                                                                           
/usr/bin/ld: obj/src/controller/python/_ChipDeviceCtrl.ChipDeviceController-ScriptBinding.cpp.o: in function `pychip_DeviceController_SetLogFilter':                                                                                                                                                                          
/home/sag/projects/project-chip/connectedhomeip-v1.2/out/python_lib/../../src/controller/python/ChipDeviceController-ScriptBinding.cpp:366:(.text.pychip_DeviceController_SetLogFilter+0x14): undefined reference to `chip::Logging::SetLogFilter(unsigned char)'  
```

But then I thought why bother, filtering shouldn't harm if not used, and the additional size/overhead is probably negligible for typical Linux systems :man_shrugging:

Comment (yunhanw-google) [reply src/platform/Linux/CHIPPlatformConfig.h:60]: yep, the macro cannot pass across gn component, i think we could enable this at default for linux, thanks

**Review by yunhanw-google — APPROVED** [review]

**Review by tehampson — APPROVED** [review]

**Pull request merged**